### PR TITLE
Rename Mixer to AudioEngine

### PR DESCRIPTION
**Pull request by M374LX**
This PR aims to rename Mixer to AudioEngine, as decided in #6089 and #5592.

At present, it leaves the "mixer" element of the configuration file as is, since backwards compatibility needs to be taken into account.

### CODE REVIEW

File: include/AudioEngineWorkerThread.h [hunk 1]
```diff
@@ -29,11 +29,11 @@
 
 #include <atomic>
 
+class AudioEngine;
```
Comment (JohannesLorenz) [inline include/AudioEngineWorkerThread.h:32]: The `#define` above is still using "Mixer"

File: include/AudioEngineProfiler.h [hunk 2]
```diff
@@ -30,11 +30,11 @@
 #include "lmms_basics.h"
 #include "MicroTimer.h"
 
-class MixerProfiler
```
Comment (JohannesLorenz) [inline include/AudioEngineProfiler.h:33]: The `#define` above is still using "Mixer".

**Review by JohannesLorenz — CHANGES_REQUESTED** [review]
Thanks for the PR!

Only 3 minor comments, plus 2 more issues:

1. In submodule `doc/wiki`, `LMMS-Architecture.md` still uses "Mixer". Can you please fix it and update the submodule pointer?
2. In `plugins/lb302/README`, there are 2 occurrences of "mixer" left.

**Comment by JohannesLorenz** [comment]
@M374LX  All OK, except the submodule:

```
Fetching submodule doc/wiki
gitFrom https://github.com/lmms/lmms.wiki
   e4288ee..d48430b  master     -> origin/master
 subfatal: remote error: upload-pack: not our ref a9dbb1c34eb925cf0adb53ac4aa366fea2aa2706
Errors during submodule fetch:
        doc/wiki
```

Did you forget to push that commit?

**Comment by M374LX** [comment]
@JohannesLorenz 
Sorry for my inexperience with Git submodules. Here is the updated ``LMMS-Architecture.md``.

[LMMS-Architecture.md](https://github.com/LMMS/lmms/files/7148370/LMMS-Architecture.md)

**Comment by JohannesLorenz** [comment]
@M374LX Can you please try something like this?

```bash
cd doc/wiki
git pull
(overwrite the current file with your copy)
git commit -am ...
git push
cd ../..
git add doc/wiki
git commit -am "Update wiki submodule"
git push
```

**Comment by M374LX** [comment]
@JohannesLorenz When running ``git push``, I get the error:
```
remote: Permission to lmms/lmms.wiki.git denied to M374LX.
fatal: unable to access 'https://github.com/lmms/lmms.wiki.git/': The requested URL returned error: 403
```

**Comment by JohannesLorenz** [comment]
OK, I can push it for you. Did you remove `More information is available at GitBook: https://docs.lmms.io/developer-guides/core/` on purpose?

**Comment by M374LX** [comment]
Here is the file with ``More information is available at GitBook: https://docs.lmms.io/developer-guides/core/`` readded. The removal was a mistake.

[LMMS-Architecture.md](https://github.com/LMMS/lmms/files/7148427/LMMS-Architecture.md)

**Comment by JohannesLorenz** [comment]
@M374LX Looks like you also removed the "Change indentation levels" commit (see https://github.com/LMMS/lmms/wiki/LMMS-Architecture/_history)

**Comment by M374LX** [comment]
Fixed.

[LMMS-Architecture.md](https://github.com/LMMS/lmms/files/7148574/LMMS-Architecture.md)

**Comment by JohannesLorenz** [comment]
Great, https://github.com/LMMS/lmms/wiki/LMMS-Architecture/_history shows that you successfully pushed to the wiki :stuck_out_tongue_winking_eye:  Now, can you please go into your submodule, do a `git push`, and then commit the submodule in your main repo?

**Comment by M374LX** [comment]
Again, the same error as in https://github.com/LMMS/lmms/pull/6127#issuecomment-917459784.

**Comment by JohannesLorenz** [comment]
Oh, I mean do a `git pull`. Sorry!

**Comment by JohannesLorenz** [comment]
Also, as soon as this PR will be finished, do you want the changes squashed into a single commit, or added to master as a chain of multiple commits?

**Comment by M374LX** [comment]
What more should I do after ``git pull``?

Squashing the commits is a better idea, as this makes the commit history cleaner.

**Comment by JohannesLorenz** [comment]
After the submodule is pulled, go back to the main repo (lmms), and

```
git add doc/wiki
git commit -m "Update wiki"
git push
```

**Comment by JohannesLorenz** [comment]
That looks correct, but I still can't pull. Probably because of the defected "Update architecture on Wiki" commit. Can you try a `git rebase -i HEAD~3`, and in the commit editor, mark

```
pick    Update architecture on Wiki
fixup   Update wiki
```

And then just `push -f` the lmms repo. That should remove the invalid submodule pointer.

**Comment by M374LX** [comment]
Possibly because of a ``git reset`` I tried previously, the commits that appear with ``git rebase -i HEAD~3`` are:
```
pick da3d082 Homebrew offers qt and qt5 packages separately.
pick d48430b Add qt6 warning/troubleshooting
pick 4feb6c9 Rename Mixer->AudioEngine (LMMS/lmms#6127)
```

**Comment by JohannesLorenz** [comment]
Oh, please abort that. I meant in the lmms repo, not in the submodule.

**Comment by JohannesLorenz** [comment]
Works! Let's merge this.

**Review by JohannesLorenz — APPROVED** [review]

**Pull request merged**